crux-agent/app/core/agents/professor.py

```
@@ -409,12 +762,29 @@ def specialist_progress_callback(current_iter: int, max_iters: int, phase: str):
             from app.settings import settings
+            # Build a stable specialist-specific job_id derived from the parent job and task
+            _parent_job_id = None
+            try:
+                _parent_job_id = get_current_job_id()
+            except Exception:
+                _parent_job_id = None
+            # Derive deterministic child job_id to avoid lock contention with professor
+            _specialist_job_id = None
+            if _parent_job_id:
+                try:
+                    _slug = re.sub(r"[^a-z0-9_]+", "", specialization.lower().replace(" ", "_"))
```

Ruff (0.12.2)

Local variable `re` referenced before assignment

(F823)